lib/galaxy/datatypes/mothur.py

```diff
@@ -0,0 +1,949 @@
+"""
+Mothur Metagenomics Datatypes
+James E Johnson - University of Minnesota
+Iyad Kandalaft <Iyad.Kandalaft@agr.gc.ca> - Agriculture and Agri-Foods Canda
+"""
+
+import logging, os, os.path, sys, time, tempfile, shutil, string, glob, re
+import galaxy.model
+from galaxy.datatypes.sniff import *
+from galaxy.datatypes.metadata import MetadataElement
+from galaxy.datatypes.data import Text
+from galaxy.datatypes.tabular import Tabular
+from galaxy.datatypes.sequence import Fasta
+from galaxy import util
+from galaxy.datatypes.images import Html
+import pkg_resources
+
+log = logging.getLogger(__name__)
+
+## Mothur Classes 
+
+class Otu( Text ):
+    file_ext = 'mothur.otu'
+    MetadataElement( name="columns", default=0, desc="Number of columns", readonly=True, visible=True, no_value=0 )
+    MetadataElement( name="labels", default=[], desc="Label Names", readonly=True, visible=True, no_value=[] )
+    def __init__(self, **kwd):
+        Text.__init__( self, **kwd )
+    def set_meta( self, dataset, overwrite = True, **kwd ):
+        if dataset.has_data():
+            label_names = set()
+            ncols = 0
+            data_lines = 0
+            comment_lines = 0
+            try:
+                with open( dataset.file_name ) as fh:
+                    for line in fh:
+                        fields = line.strip().split('\t')
+                        if len(fields) >= 2: 
+                            data_lines += 1
+                            ncols = max(ncols,len(fields))
+                            label_names.add(fields[0])
+                        else:
+                            comment_lines += 1
+                    # Set the discovered metadata values for the dataset
+                    dataset.metadata.data_lines = data_lines
+                    dataset.metadata.columns = ncols
+                    dataset.metadata.labels = list( label_names )
+                    dataset.metadata.labels.sort()
+            except:
+                pass
+
+    def sniff( self, filename ):
+        """
+        Determines whether the file is a otu (operational taxonomic unit) format
+        """
```

Comment:
You could also think about some doc-tests like this one: https://github.com/galaxyproject/galaxy/blob/dev/lib/galaxy/datatypes/xml.py#L31